transformations/style_paraphraser/paraphraser_helpers/style_paraphraser.py

```diff
@@ -0,0 +1,394 @@
+                    F.softmax(filtered_logits, dim=-1), num_samples=1
+                )
+            )
+            generated = torch.cat((generated, next_token), dim=1)
+            segments = torch.cat((segments, segments[:, -1:]), dim=1)
+
+            for batch_elem in range(batch_size):
+                if next_token[batch_elem].item() == eos_token_id:
+                    eos_emitted[batch_elem] = True
+
+            if length is None and all(eos_emitted):
+                break
+    return generated
+
+
+def _score_fn(x, length_normalize):
+    if length_normalize:
+        return x["score"] / len(x["sequence"])
+    else:
+        return x["score"]
+
+
+class Instance(object):
+    """
+    Instance of a sentence to generate a paraphrase for.
+
+    Args:
+        args: Standard, default arguments to use.
+        config : configurations specified.
+        instance_dict : dictionary containing the sentence to paraphrase.
+
+    """
+
+    def __init__(self, args, config, instance_dict):
+        self.dict = instance_dict
+        self.args = args
+        self.config = config
+        self.truncated = False
+        self.sent1_tokens = np.array(
+            instance_dict["sent1_tokens"], dtype=np.int32
+        )
+        self.sent2_tokens = np.array(
+            instance_dict["sent2_tokens"], dtype=np.int32
+        )
+        self.init_context_size = config["max_prefix_length"] + 1
+
+    def preprocess(self, tokenizer):
+        self.truncate()
+        self.build_sentence(tokenizer)
+        self.build_label(tokenizer)
+        self.build_segment(tokenizer)
+
+    def truncate(self):
+        config = self.config
+        max_prefix_length = config["max_prefix_length"]
+        max_suffix_length = config["max_suffix_length"]
+        if len(self.sent1_tokens) > max_prefix_length:
+            self.truncated = True
+            self.sent1_tokens = self.sent1_tokens[:max_prefix_length]
+        if len(self.sent2_tokens) > max_suffix_length:
+            self.truncated = True
+            self.sent2_tokens = self.sent2_tokens[:max_suffix_length]
+
+    def build_sentence(self, tokenizer):
+        tokens_to_pad = self.config["max_prefix_length"] - len(
+            self.sent1_tokens
+        )
+        self.sent_prefix = np.pad(
+            self.sent1_tokens,
+            (tokens_to_pad, 0),
+            constant_values=tokenizer.pad_token_id,
+        )
+
+        self.sent_suffix = self.right_padding(
+            np.append(self.sent2_tokens, tokenizer.eos_token_id),
+            tokenizer.pad_token_id,
+            self.config["max_suffix_length"] + 1,
+        )
+        self.sentence = np.concatenate(
+            [self.sent_prefix, [tokenizer.bos_token_id], self.sent_suffix]
+        )
+
+    def right_padding(self, data, pad_token, total_length):
+        tokens_to_pad = total_length - len(data)
+        return np.pad(data, (0, tokens_to_pad), constant_values=pad_token)
+
+    def build_label(self, tokenizer):
```

Comment:
It seems that the labels are not used for inference. There are only 3 occurrences of `.label`, and they are all inside this function.

Reply:
This is great! I tried to remove as much redundant code of the original as possible, but obviously I missed this and the previous part :) Thank you!

Reply:
Fixed!